.github/workflows/test.yml

```diff
@@ -52,19 +51,29 @@ jobs:
         path: |
           ${{ env.CI_IMAGE_CACHE }}
         key: ${{ github.run_id }}-image-cache-master
-
+    # if CI_IMAGE_MASTER_TAR isn't in cache, try pulling it and saving to the cache rather 
+    # than building, resort back to building if the cache isn't populated and 
+    # pulling the image fails.
     - name: Check if master image build is needed
       id: is_master_image_build_needed
-      continue-on-error: true
+      continue-on-error: false
       run: |
         set -x
-        if [ -f ${CI_IMAGE_CACHE}/${CI_IMAGE_MASTER_TAR}.gz ]; then
-            mkdir -p ${CI_DIST_IMAGES_OUTPUT}
-            cp ${CI_IMAGE_CACHE}/${CI_IMAGE_MASTER_TAR}.gz ${CI_DIST_IMAGES_OUTPUT}/${CI_IMAGE_MASTER_TAR}.gz
-            gunzip ${CI_DIST_IMAGES_OUTPUT}/${CI_IMAGE_MASTER_TAR}.gz
+        if [ -f ${CI_IMAGE_CACHE}${CI_IMAGE_MASTER_TAR}.gz ]; then
+            cp ${CI_IMAGE_CACHE}/${CI_IMAGE_MASTER_TAR}.gz ${CI_IMAGE_MASTER_TAR}.gz
+            gunzip ${CI_IMAGE_MASTER_TAR}.gz
+            echo "::set-output name=LOAD_MASTER_IMAGE_TO_CACHE::false"
```

Comment:
nit:
It would have been much easier for me to understand if we had an `MASTER_IMAGE_RESTORED_FROM_CACHE` instead of `LOAD_MASTER_IMAGE_TO_CACHE`
`MASTER_IMAGE_RESTORED_FROM_CACHE` would be set to false by default and to true within this if clause.

Reply:
Ack changed the variables to `MASTER_IMAGE_RESTORED_FROM_GHCR` and `MASTER_IMAGE_RESTORED_FROM_CACHE`